src/v/utils/vint.h

```diff
@@ -53,7 +53,8 @@ struct var_decoder {
 
 template<typename Range>
 inline std::pair<uint64_t, size_t>
-deserialize(Range&& r, uint8_t limit) noexcept {
+deserialize(Range&& r, size_t max_length) noexcept {
+    auto limit = ((max_length - 1) * 7);
```

Comment:
surprised this didn't give a shadowing warning about the name? anyway, seems fine.

Reply:
About `max_length`? Yeah I can update.

Reply:
Fixed